scripts/globals/helm.lua

```diff
@@ -1430,6 +1431,31 @@ dsp.helm.onTrade = function(player, npc, trade, helmType, csid)
             npcUtil.giveKeyItem(player, dsp.ki.RAINBOW_BERRY)
         end
 
+        local amkChance = 20
+        if 
+            player:getCurrentMission(AMK) == dsp.mission.id.amk.WELCOME_TO_MY_DECREPIT_DOMICILE and
+            broke ~= 1
+        then
+            if
+                helmType == dsp.helm.type.MINING and
+                not player:hasKeyItem(dsp.ki.STURDY_METAL_STRIP) and
+                dsp.expansionZone.ORIGINAL_ROTZ[zoneId] and math.random(0,99) < amkChance
```

Comment:
I really dislike doing `0,99` in math.random calls. I would like to see this conditional altered to just used `math.random(100)` instead. That will roll 1-100 which means it won't start at zero and is more intuitive in my opinion for things that are being treated as a % chance. You'll have to alter the comparison to account for that shift by 1.